### PR TITLE
Add field selector support to delete, label, annotate

### PR DESCRIPTION
**Pull request by liggitt**
**What this PR does / why we need it**:

Builds out field selector support for more bulk commands

**Release note**:

```release-note
kubectl now supports --field-selector for `delete`, `label`, and `annotate`
```

### CODE REVIEW

**Comment by liggitt** [comment]
@kubernetes/sig-cli-pr-reviews

**Comment by krmayankk** [comment]
@liggitt did you mean label selector or field selector. Is my understanding correct that now you would be able to annotate for e.g. multiple pods which match a label ?

**Comment by liggitt** [comment]
> @liggitt did you mean label selector or field selector. Is my understanding correct that now you would be able to annotate for e.g. multiple pods which match a label ?

Field selector. You can already annotate pods which match a label:
```
kubectl annotate pods --selector somelabel=somevalue newannotation=newvalue
```

**Review by ericchiang — COMMENTED** [review]
Need to make changes to LabelOptions in pkg/kubectl/cmd/label.go

**Comment by liggitt** [comment]
> Need to make changes to LabelOptions in pkg/kubectl/cmd/label.go

done

**Comment by liggitt** [comment]
rebased. @soltysh @juanvallejo PTAL at delete changes

File: pkg/kubectl/cmd/delete_flags.go [hunk 1]
```diff
@@ -113,7 +121,12 @@ func (f *DeleteFlags) ToOptions(out, errOut io.Writer) *DeleteOptions {
 
 func (f *DeleteFlags) AddFlags(cmd *cobra.Command) {
 	f.FileNameFlags.AddFlags(cmd)
-
+	if f.LabelSelector != nil {
+		cmd.Flags().StringVarP(f.LabelSelector, "selector", "l", *f.LabelSelector, "Selector (label query) to filter on, not including uninitialized ones.")
+	}
+	if f.FieldSelector != nil {
+		cmd.Flags().StringVarP(f.FieldSelector, "field-selector", "", *f.FieldSelector, "Selector (field query) to filter on, supports '=', '==', and '!='.(e.g. --field-selector key1=value1,key2=value2). The server only supports a limited number of field queries per type.")
```
Comment (dixudx) [inline pkg/kubectl/cmd/delete_flags.go]: Can use `cmd.Flags().StringVar()` here for missing shorthand letter.

**Comment by juanvallejo** [comment]
"delete" changes lgtm

**Review by juanvallejo — APPROVED** [review]

**Pull request merged**